File: problems/13333.py
# https://www.acmicpc.net/problem/13333

from sys import stdin

input = stdin.readline

N = int(input())
indice = list[int](map[int](int, input().split()))
indice.append(0)
indice.sort(reverse=True)

ans = 0
for k in range(1, N + 1):
    if indice[k - 1] >= k and indice[k] <= k:
        ans = k

print(ans)

File: problems/13333.kt
// https://www.acmicpc.net/problem/13333

fun main() {
    val N:Int = readln().toInt()
    val indice:MutableList<Int> = readln().split(" ").map { it.toInt() }.toMutableList()
    indice.add(0)
    indice.sortDescending()

    var ans:Int = 0
    for (k in 1..N) {
        if (indice[k - 1] >= k && indice[k] <= k) {
            ans = k
        }
    }

    println(ans)
}

